### DISCUSSION TITLE
Help with Discord webhooks, entering more ADMINLIST_IDS (RESOLVED) and Backups aren't happening (RESOLVED)

### DISCUSSION THREAD

**Discussion by Xevichr** [lloesche/valheim-server-docker#680] [Q&A]
See my current docker run all the way below. 

Currently the webhook has the following output when restarting at 6 AM:
_Shutting down server.
Server started successfully.
Shutdown completed.
Starting server. Please wait 5 minutes._

The exact same when manually stopping the container.

**(RESOLVED, kinda? See Edit "Discord Webhooks"):** So, why would it say in the middle of restarting that the "Server started successfully."? Anyone know how I can prevent this, and how I can make this sent when it is actually relevant? This is the only point where it is ever being sent.

**Edit (Discord Webhooks):** In my current version, I just flat out removed the webhook messages that never worked, such as an announcement happening before server restart.

Besides that, I also removed the message that sent _"Server started successfully."_, because it more or less never sent when it should've been. _"Starting server. Please wait 5 minutes."_ actually does get sent everytime, and effectively, for your average user, means the same thing anyway.

**Status for the Discord webhooks:** 
What I would still like:
Message for player disconnects
Message for player death
Message for random event **(RESOLVED, in current version below for those wondering)** 

**The ADMINLIST: How can I put several ADMINLIST_IDS? (RESOLVED):**
Do I just say:
```
-e ADMINLIST_IDS="1234
1234
1234" \
```

**Edit (ADMINLIST):** Yes. You do. lol. This does start a new line.

**Backups not happening when scheduled with cronjob (RESOLVED):**

**Edit (Backups):** I had a few questions regarding backups, but I figured that out myself. Basically having `BACKUPS_IF_IDLE` set to false prevented backups from being made when I actually scheduled them to be made with cron. Nobody played around the time of the cronjob, so `BACKUPS_IF_IDLE` deemed it not worth saving.

**EDIT (current version):**
```
docker run -d \
    --name valheim-server \
    --cap-add=sys_nice \
    --stop-timeout 120 \
    -p 2456-2457:2456-2457/udp \
    -v $HOME/valheim-server/config:/config \
    -v $HOME/valheim-server/data:/opt/valheim \
    -e SERVER_NAME="___" \
    -e WORLD_NAME="___" \
    -e SERVER_PASS="___" \
    -e SERVER_PUBLIC="false" \
    -e TZ="Europe/Amsterdam" \
    -e BACKUPS_CRON="0 5 * * *" \
    -e BACKUPS_DIRECTORY="/config/backups" \
    -e BACKUPS_MAX_COUNT="0" \
    -e BACKUPS_MAX_AGE="7" \
    -e BACKUPS_IF_IDLE="true" \
    -e RESTART_CRON="0 6 * * *" \
    -e UPDATE_IF_IDLE=true \
    -e ADMINLIST_IDS="___
___" \
    -e DISCORD_WEBHOOK="https://discord.com/api/webhooks/___" \
    -e VALHEIM_LOG_FILTER_CONTAINS_Spawned="Got character ZDOID from" \
    -e ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned='{ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="Player $l spawned into the world"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }' \
    -e VALHEIM_LOG_FILTER_CONTAINS_RandomEvent="Random event set:" \
    -e ON_VALHEIM_LOG_FILTER_CONTAINS_RandomEvent='{ read l; l=${l//*Random event set:/}; l=${l// :/}; l=${l#* }; msg="Random event: $l"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }' \
    -e DISCORD_PRERUNMESSAGE="Starting server. Please wait 3-5 minutes." \
    -e PRE_SERVER_RUN_HOOK='curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"content\":\"$DISCORD_PRERUNMESSAGE\"}" "$DISCORD_WEBHOOK"' \
    -e DISCORD_PRESHUTDOWNMESSAGE="Shutting down server (Manual shutdown, scheduled shutdown at 6 AM CEST/CET or error)." \
    -e PRE_SERVER_SHUTDOWN_HOOK='curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"content\":\"$DISCORD_PRESHUTDOWNMESSAGE\"}" "$DISCORD_WEBHOOK"' \
    -e DISCORD_POSTSHUTDOWNMESSAGE="Shutdown completed." \
    -e POST_SERVER_SHUTDOWN_HOOK='curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"content\":\"$DISCORD_POSTSHUTDOWNMESSAGE\"}" "$DISCORD_WEBHOOK"' \
    -e DISCORD_PRERESTARTMESSAGE="Restarting Valheim server now (Server updating or Container restart)." \
    -e PRE_RESTART_HOOK='curl -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$DISCORD_PRERESTARTMESSAGE\"}" "$DISCORD_WEBHOOK"' \
    --restart unless-stopped \
    ghcr.io/lloesche/valheim-server
```

**Comment by fizch** [comment]
As far as the admin list goes, you can put as many admins as you would like in the adminlist.txt file. That is located in the config path. You have it defined as $HOME/valheim-server/config.

Unfortunately, I cannot really help with the discord web hook issues. I am still working through issues of my own there.

**Reply by Xevichr** [reply]
Thanks for the response.

I am aware that I can manually edit the adminlist like this, but I prefer to put multiple by default when I run the container. Because the adminlist I manually changed will be overwritten as soon as I reinitiate the container completely at some point. 

Because of the questions I still have regarding the Discord webhook, I will completely reinitiate the container quite a few times over the course of the next days, so having to do anything regarding this manually will just be annoying.

I looked into it and I will try to seperate the values with commas, which is apperently the common way it is done. I'll update my post if it works out.

**Comment by Xevichr** [comment]
Still looking for help with Discord webhook:
Message for player disconnects
Message for player death

My Discord webhook contains:
Message for player spawn (However, happens when they get online and when they respawn after dying)
Message for random event happening (Such as surtlings attacking your base)
Message for server starting
Message for server shutdown
Message for succesful server shutdown
Message for restarting the server

Hope it can help someone in the future. 

For the backup issues I had and adding multiple Admin IDS by default, read the whole post.

**Comment by jasontibbitts** [comment]
I noticed that death notices look like spawn events, but the location is always 0:0.  I'll include what I'm using below, but note that I'm running in a Podman quadlet so you would need to adapt the syntax and quoting.

```
Environment="VALHEIM_LOG_FILTER_REGEXP_Croaked=Got character ZDOID from.*: 0:0"
Environment='ON_VALHEIM_LOG_FILTER_REGEXP_Croaked={ read l; l=$${l//*ZDOID from /}; l=$${l// :*/}; msg=\"Player $$l just croaked!\"; curl -sfSL -X POST -H \"Content-Type: application/json\" -d \"{\\\"username\\\":\\\"Valheim\\\",\\\"content\\\":\\\"$$msg\\\"}" "$$DISCORD_WEBHOOK"; }'

Environment="VALHEIM_LOG_FILTER_REGEXP_Spawned=Got character ZDOID from.*: -?[1-9]"
Environment='ON_VALHEIM_LOG_FILTER_REGEXP_Spawned={ read l; l=$${l//*ZDOID from /}; l=$${l// :*/}; msg=\"Player $$l spawned.\"; curl -sfSL -X POST -H \"Content-Type: application/json\" -d \"{\\\"username\\\":\\\"Valheim\\\",\\\"content\\\":\\\"$$msg\\\"}" "$$DISCORD_WEBHOOK"; }'
```
